mm/mm_heap/mm_extend.c

```diff
@@ -77,7 +77,7 @@ void mm_extend(FAR struct mm_heap_s *heap, FAR void *mem, size_t size,
 
   /* Take the memory manager semaphore */
 
-  DEBUGVERIFY(mm_takesemaphore(heap));
+  DEBUGASSERT(mm_takesemaphore(heap));
```

Comment:
but DEBUGASSERT become empty when CONFIG_DEBUG_ASSERTIONS isn't defined

Reply:
Both define are not empty when CONFIG_DEBUG_ASSERTIONS isn't defined
`#  define DEBUGASSERT(f) ((void)(1 || (f)))`
`#  define DEBUGVERIFY(f) ((void)(f))`

Reply:
Yes, but the result is no-op, since this first one will skip mm_takesemaphore when CONFIG_DEBUG_ASSERTIONS isn't defined.

Reply:
We can't use `DEBUGASSERT`. We need to evaluate some other way

Reply:
Got it. 
How about change the mm_takesemaphore return value to 0 if success, and negative if fails.

Reply:
could be

Reply:
Another method:
```suggestion
ret = mm_takesemaphore(heap)
DEBUGASSERT(ret);
```
what's your opinion? @xiaoxiang781216 @pkarashchenko

Reply:
Could be, but you need handle the unused local variable warning.

Reply:
or use ? : like this:
```
DEBUGVERIFY(mm_takesemaphore(heap) ? 0 : -EINVAL);
```
or add DEBUGVERIFYBOOL

Reply:
> Another method:
> 
> what's your opinion? @xiaoxiang781216 @pkarashchenko

This and add `unused_data` to `ret` definition.
https://github.com/apache/incubator-nuttx/blob/1b9c013dade651d6998c5b9463e4788282b12e4f/include/nuttx/compiler.h#L228

Reply:
The `ret` is referenced wether CONFIG_DEBUG_ASSERTIONS defined or not, so, unused_data is not required. I have build with enable and disable CONFIG_DEBUG_ASSERTIONS, there is no addition warnings.
```
#define ASSERT(f)        do { if (!(f)) PANIC(); } while (0)
# define DEBUGASSERT(f) ((void)(1 || (f)))
```